### ISSUE TITLE
AI Image Description Throws Error on Consecutive Recognitions

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#19058]
### Brief summary

When attempting to use NVDA's AI image description feature multiple times in rapid succession, a `TypeError` is thrown, preventing the recognition from completing.

### Steps to reproduce

1.  Focus on an object or image that can be recognized by the AI image description feature.
2.  Press the `NVDA+Windows+,` to get image description.
3.  Immediately after the first recognition command is issued, press the same shortcut key combination again, one or more times, in quick succession.

### Actual behavior

When the get image description command is triggered multiple times in rapid succession, NVDA's log shows a `TypeError: cannot unpack non-iterable NoneType object`.

The following traceback is consistently observed in the NVDA log:

```log
ERROR - scriptHandler.executeScript (18:10:59.196) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
```

### Expected behavior

NVDA should handle consecutive requests for image description gracefully.

### NVDA logs, crash dumps and other attachments

ERROR - scriptHandler.executeScript (18:10:59.196) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
ERROR - scriptHandler.executeScript (18:11:00.313) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
ERROR - scriptHandler.executeScript (18:11:00.856) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
ERROR - scriptHandler.executeScript (18:11:00.897) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
ERROR - scriptHandler.executeScript (18:11:00.948) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object
ERROR - scriptHandler.executeScript (18:11:00.984) - MainThread (3624):
error executing script: <bound method GlobalCommands.script_runCaption of <globalCommands.GlobalCommands object at 0x0000000008538D70>> with gesture 'NVDA+Windows+,'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "globalCommands.pyc", line 5108, in script_runCaption
  File "_localCaptioner\imageDescriber.pyc", line 114, in runCaption
  File "_localCaptioner\imageDescriber.pyc", line 43, in _screenshotNavigator
TypeError: cannot unpack non-iterable NoneType object

### NVDA type

portable copy

### NVDA version

2026.1 Alpha

### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

### Windows version

Windows 10 22H2 (AMD64) build 19045.6396

### Name and version of other software in use when reproducing the issue

Windows desktop icons

### Other information about your system

None

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I do not have any NVDA add-ons installed

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by cary-rowen** [comment]
When addressing this bug, please avoid implementing a simple debouncing or throttling strategy (i.e., ignoring rapid inputs) as the primary solution.

The recognition engine itself should be robust enough to handle a continuous stream of requests. This is critical for potential future enhancements, such as an automatic image description feature. For example, a future implementation might continuously describe objects on screen as the user navigates, which would rely on the ability to process a rapid and uninterrupted flow of recognition requests without failure.

**Comment by seanbudd** [comment]
@tianzeshi-study - can you look into this please?

**Comment by seanbudd** [comment]
@tianzeshi-study - Hi, could you confirm if you are able to look into this? If you are unable to, that is okay, just let us know

**Comment by tianzeshi-study** [comment]
Oh, Sorry, I forgot to reply earlier. I haven’t been able to reproduce this issue so far, so I’m waiting for more information.

According to the logs, the issue seems to be caused by the screenshot function being called rapidly; the exact cause is still unclear, but it likely isn’t a problem with the actual image description parsing/processing.

**Comment by seanbudd** [comment]
Do you not run into an issue when spamming the script to get an image description? what happens if you run `_screenshotNavigator` in a for/while loop, from within the NVDA python console?
@cary-rowen - can you still reproduce this issue in the latest alphas?
